### PR TITLE
fix overblur in AA

### PR DESCRIPTION
**Pull request by ducha-aiki**
Fixes https://github.com/kornia/kornia/discussions/1613

After the check, I agree with @[scott-vsi](https://github.com/scott-vsi), so I have switched the sigma calculation to skimage one 

Before:
image

After the change:
image

So that is a good direction. It also should speed-up things, because smaller sigma requires smaller kernel sizes

No AA (just for comparison)
image

#### Type of change

- [ ] 📚  Documentation Update
- [ ] 🧪 Tests Cases
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 This change requires a documentation update

#### Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

File: kornia/geometry/transform/affwarp.py [hunk 1]
```diff
@@ -581,12 +581,14 @@ def resize(
...
-        sigmas = (max(factors[0], 1.0), max(factors[1], 1.0))
+        # Taken from skimage https://github.com/scikit-image/scikit-image/blob/skimage/transform/_warps.py#L171
+        sigmas = (max((factors[0] - 1.0) / 2.0, 0.001),
+                  max((factors[1] - 1.0) / 2.0, 0.001))
 
         # Now kernel size. Good results are for 3 sigma, but that is kind of slow. Pillow uses 1 sigma
         # https://github.com/python-pillow/Pillow/blob/master/src/libImaging/Resample.c#L206
         # But they do it in the 2 passes, which gives better results. Let's try 2 sigmas for now
-        ks = int(2.0 * 2 * sigmas[0]), int(2.0 * 2 * sigmas[1])
+        ks = int(max(2.0 * 2 * sigmas[0], 3)), int(max(2.0 * 2 * sigmas[1], 3))
```
Comment (ducha-aiki) [inline kornia/geometry/transform/affwarp.py:591]: This is needed to avoid getting kernel size 0 or 1

File: kornia/geometry/transform/affwarp.py [hunk 2]
```diff
@@ -581,12 +581,14 @@ def resize(
 
     if antialias:
         # First, we have to determine sigma
-        sigmas = (max(factors[0], 1.0), max(factors[1], 1.0))
+        # Taken from skimage https://github.com/scikit-image/scikit-image/blob/skimage/transform/_warps.py#L171
```
Comment (scott-vsi) [inline kornia/geometry/transform/affwarp.py:584]:
```suggestion
        # Taken from skimage: https://github.com/scikit-image/scikit-image/blob/v0.19.2/skimage/transform/_warps.py#L171
```

File: kornia/geometry/transform/affwarp.py [hunk 3]
```diff
@@ -581,12 +581,14 @@ def resize(
 
     if antialias:
         # First, we have to determine sigma
-        sigmas = (max(factors[0], 1.0), max(factors[1], 1.0))
+        # Taken from skimage: https://github.com/scikit-image/scikit-image/blob/v0.19.2/skimage/transform/_warps.py#L171
+        sigmas = (max((factors[0] - 1.0) / 2.0, 0.001),
+                  max((factors[1] - 1.0) / 2.0, 0.001))
```
Comment (edgarriba) [inline kornia/geometry/transform/affwarp.py:586]: With 0.001 instead of 0. As scikt suggests  ?

**Review by edgarriba — APPROVED** [review]
LGTM. Just one question, I see that our logic is to apply antialiasing if and only if user specifies and is down sampling. Is doable to have as scikit suggests to apply always by default if down sampling happens except when user pass false ?

File: kornia/geometry/transform/affwarp.py (hunk 3 above)
Comment (ducha-aiki) [reply kornia/geometry/transform/affwarp.py:586]: There is no sense is sigma 0

Comment (ducha-aiki) [reply kornia/geometry/transform/affwarp.py:586]: To apply "by default" for upsampling also does not make sense :)

**Pull request merged**